.eslintrc.js

```diff
@@ -28,123 +32,50 @@ module.exports = {
-        '@typescript-eslint/no-unused-expressions': 0,
+        '@typescript-eslint/no-var-requires': 0,
+        '@typescript-eslint/no-explicit-any': 0,
+        '@typescript-eslint/no-inferrable-types': 0,
+        '@typescript-eslint/no-non-null-assertion': 0,
+        '@typescript-eslint/prefer-nullish-coalescing': 0,
+        '@typescript-eslint/consistent-indexed-object-style': 0,
+        '@typescript-eslint/consistent-type-assertions': 0,
+        '@typescript-eslint/consistent-type-definitions': 0,
+        '@typescript-eslint/no-empty-interface': 0,
+
         // eslint-plugin-import
-        'import/extensions': 0,
-        'import/no-cycle': 1,
-        'import/no-extraneous-dependencies': 0,
-        'import/no-unresolved': 0,
-        'import/prefer-default-export': 0,
-        // eslint-plugin-react
-        // TODO: [TS DEBT] should remove below config once application converted to TS
-        'react/default-props-match-prop-types': 0,
-        'react/require-default-props': 0,
-        '@typescript-eslint/no-useless-constructor': 0,
-        // eslint-plugin-react
-        'react/destructuring-assignment': 0,
-        'react/button-has-type': 0,
-        'react/forbid-prop-types': 0,
-        'react/jsx-curly-newline': 0,
-        'react/no-unused-prop-types': 1,
-        'react/react-in-jsx-scope': 0,
-        'react/jsx-no-bind': 0,
-        'react/jsx-no-target-blank': 0,
-        'react/jsx-props-no-spreading': 0,
-        'react/no-deprecated': 1,
-        'react/no-array-index-key': 0,
-        'react/prefer-stateless-function': 0,
-        'react/sort-comp': 0,
-        'react/state-in-constructor': 0,
-        'react/static-property-placement': 0,
-        'react/function-component-definition': 0,
-        'react/jsx-no-useless-fragment': 0,
-        //  eslint-plugin-jsx-a11y
-        'jsx-a11y/click-events-have-key-events': 1,
-        'jsx-a11y/mouse-events-have-key-events': 1,
-        'jsx-a11y/label-has-for': [
-          2,
-          {
-            components: ['Label'],
-            required: {
-              every: ['id'],
-            },
-            allowChildren: false,
-          },
-        ],
-        'jsx-a11y/no-static-element-interactions': 0,
-        'jsx-a11y/no-noninteractive-element-interactions': 1,
-        //  eslint-plugin-unicorn
-        'unicorn/filename-case': 0,
-        'unicorn/no-null': 0,
-        'unicorn/no-useless-undefined': 0,
-        'unicorn/prefer-module': 0,
-        'unicorn/prevent-abbreviations': 0,
-        'unicorn/prefer-node-protocol': 0,
-        'unicorn/import-style': [
-          2,
-          {
-            styles: {
-              path: {
-                named: true,
-              },
-            },
-          },
-        ],
-        'unicorn/consistent-destructuring': 0,
-        // INFO: Turned off due to src/internal-server/database/factory.js
-        'unicorn/no-empty-file': 0,
-        // eslint-plugin-prettier
-        'prettier/prettier': 1,
```

Comment:
I merged these TS-specific rules with the ones on root level. It makes it easier to maintain the ESLint setup and we shouldn't need to have different rule configs for JS vs TS